public/app/plugins/datasource/cloudwatch/components/QueryEditor.tsx

```diff
@@ -105,7 +92,8 @@ export class QueryEditor extends PureComponent<Props, State> {
   }
 
   render() {
-    const { query, datasource, onChange, onRunQuery, data } = this.props;
+    const { datasource, onChange, onRunQuery, data } = this.props;
+    const query = normalizeQuery(this.props.query);
```

Comment:
I like this approach, any comments on this @davkal ?

Reply:
So seems like this makes some tests fail, I think the issue is that this creates copy of props and `this.props` is not normalised while we pass that at least https://github.com/grafana/grafana/pull/21163/files#diff-3d15a6bca92bf6d3cab962e101acb86eR67.